server/jetstream_versioning.go

```diff
@@ -47,13 +53,29 @@ func setStaticStreamMetadata(cfg *StreamConfig, prevCfg *StreamConfig) {
 	cfg.Metadata[JSRequiredLevelMetadataKey] = strconv.Itoa(requiredApiLevel)
 }
 
+// setDynamicStreamMetadata adds dynamic fields into the (copied) metadata.
+func setDynamicStreamMetadata(cfg StreamConfig) StreamConfig {
```

Comment:
Doing the pass by value / copy here twice again, in and out of this function.

Reply:
Changed to use pointers